src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/ProjectSystem/DefaultRazorDynamicFileInfoProviderTest.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Moq;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.ProjectSystem
+{
+    public class DefaultRazorDynamicFileInfoProviderTest
+    {
+        public DefaultRazorDynamicFileInfoProviderTest()
+        {
+            DocumentServiceFactory = Mock.Of<DocumentServiceProviderFactory>();
+        }
+
+        private DocumentServiceProviderFactory DocumentServiceFactory { get; }
+
+        [Fact]
+        public void UpdateLSPFileInfo_UnknownFile_Noops()
+        {
+            // Arrange
+            var provider = new DefaultRazorDynamicFileInfoProvider(DocumentServiceFactory);
+            provider.Updated += (sender, args) => throw new XunitException("Should not have been called.");
+
+            // Act & Assert
+            provider.UpdateLSPFileInfo(new Uri("C:/this/does/not/exist.razor"), Mock.Of<DynamicDocumentContainer>());
+        }
+
+        // Can't currently add any more tests because of IVT restrictions from Roslyn
```

Comment:
Just when I thought "He didn't add tests for SuppressDocument. I got him!", I see this 😆